### PR TITLE
Early out for unequal NFC counts in String ==

### PR DESCRIPTION
**Pull request by Catfish-Man**

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please benchmark

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Review by stephentyrone — APPROVED** [review]
This LGTM, but I think a comment explaining why this makes sense to do here is in order.

File: stdlib/public/core/StringComparison.swift [hunk 1]
```diff
@@ -104,6 +104,9 @@ private func _stringCompareFastUTF8(
   bothNFC: Bool
 ) -> Bool {
   if _fastPath(bothNFC) {
+    if expecting == .equal && utf8Left.count != utf8Right.count {
```
[the remark this answers is not in the document]
Comment (stephentyrone) [reply stdlib/public/core/StringComparison.swift:107]: Yeah, I wanted to sink it into _binaryCompare too, but that has to produce < = >, so it has to inspect up to the first mismatch. This seems like a good fix.

**Comment by Catfish-Man** [comment]
@swift-ci please test and merge

**Comment by Catfish-Man** [comment]
Transient failure

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test and merge

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Pull request merged**